lib/instrumentation/async-hooks.js

```diff
@@ -90,11 +90,11 @@ module.exports = function (ins) {
 
   function before (asyncId) {
     const span = activeSpans.get(asyncId)
-    if (span) {
+    if (span && !span.ended) {
       span.sync = false
```

Comment:
I see here that we are checking to see if the span had ended before marking is as an asynchronous span.  To repeat discussions we've had elsewhere, this is the correct behavior. 👍  This `before` handler will fire before a specific asyncronous resource's callback runs. If a span has ended before this callback runs, it means the span completed without "crossing an asynchronous boundary", and is a sync span.